src/relayserver/RelayServer.ts

```diff
@@ -347,22 +346,25 @@ export class RelayServer extends EventEmitter {
       }, this.config.readyTimeout)
     }
 
-    return this.contractInteractor.getBlock('latest')
-      .then(
-        block => {
-          if (block.number > this.lastScannedBlock) {
-            return this._workerSemaphore.bind(this)(block.number)
-          }
+    return new Promise<void>((resolve, reject) => {
```

Comment:
Basically this was wrong since you can't return the result of then on a promise, the result of then is void and not Promise<void>, the tsc traspiler was complaining about it so this was fixed by the change i made. Make sense to you?

Reply:
yup! thanks